dspace-api/src/test/java/org/dspace/AbstractUnitTest.java

```diff
@@ -122,9 +119,6 @@ public void init() {
 
             context.restoreAuthSystemState();
 
-            // Ensure all tests run with Solr indexing disabled
-            disableSolrIndexing();
```

Comment:
All the changes to `AbstractUnitTest` appear to not be necessary now that you switched `DiscoveryIT` to extend `AbstractIntegrationTestWithDatabase` (which has Solr enabled by default).  I tested locally reverting these changes to `AbstractUnitTest` and `DiscoveryIT` still succeeds.

Therefore, I'd recommend we remove these changes, as technically Solr *should be disabled* for Unit Tests (which don't need it).  It's only supposed to be enabled for Integration Tests.

Reply:
We tried doing this, but when we run the following command locally:

> mvn install -B -V -P-assembly -Pcoverage-report -DskipIntegrationTests=false -Denforcer.skip=true -Dcheckstyle.skip=true -Dlicense.skip=true -Dxml.skip=true

We still get an exception:

`
[ERROR] deleteWorkspaceItemSolrRecordAfterDeletionFromDbTest(org.dspace.discovery.DiscoveryIT)  Time elapsed: 0.307 s  <<< FAILURE!
java.lang.AssertionError: expected:<1> but was:<0>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at org.junit.Assert.assertEquals(Assert.java:631)
	at org.dspace.discovery.DiscoveryIT.deleteWorkspaceItemSolrRecordAfterDeletionFromDbTest(DiscoveryIT.java:52)
`

This is because there are integration tests that extend from the "AbstractUnitTest", if one of these is run before the DiscoveryIT tests than the discovery test will crash (so it can depend on the system if that happens).

Reply:
this is related to the issue https://github.com/DSpace/DSpace/issues/3055 I guess that there are some *IT that extends the wrong hierarchy leaving the test env dirty. We need to found them, it is not a solution to make more complex and wrong the current solution.
I guess that a maximum of 8 hours should be sufficient to cleanup enough test to get this work properly and if you needed (@tdonohue ) my team can work on that.
That said, for this PR I suggest to revert the change as by Tim request and flag the test as Ignored opening a dedicated issue

Reply:
@KevinVdV : I agree with @abollini 's suggestion here.  We must revert these changes to `AbstractUnitTest`, but if they cause issues with the tests added here, we can _temporarily_ disable the new tests (we already know they work) until we can figure out which other UnitTests are improperly adding content into Solr.